dataikuapi/dss/ml.py

```diff
@@ -1490,13 +1512,471 @@ def __init__(self, raw_settings, hyperparameter_search_params):
+        feature's settings, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the preprocessing settings for an external feature
+        :rtype: dict
+        """
+        return self.get_feature_preprocessing(feature_name)
+
+    def get_timestep_params(self):
+        """
+        Gets the time step parameters for the time series forecasting task. This returns a reference to the
+        time step parameters, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the time step parameters
+        :rtype: dict
+        """
+        return self.mltask_settings["timestepParams"]
+
+    def set_timestep_params(self, timeunit=None, n_timeunits=None, end_of_week_day=None):
+        """
+        Sets the time step parameters for the time series forecasting task.
+
+        :param timeunit: time unit for forecasting. Valid values are: MILLISECOND, SECOND, MINUTE, HOUR, DAY,
+                         BUSINESS_DAY, WEEK, MONTH, QUARTER, HALF_YEAR, YEAR
+        :type timeunit: str
+        :param n_timeunits: number of time units within a time step
+        :type n_timeunits: int
+        :param end_of_week_day: only useful for the week time unit. Valid values are: 1 (Sunday), 2 (Monday), ..., 7 (Saturday)
+        :type end_of_week_day: int
+        :return:
+        """
+        timestep_params = self.get_timestep_params()
+        if timeunit is not None:
+            assert timeunit in self.TIMEUNITS, "Invalid timeunit, should be in {}".format(self.TIMEUNITS)
+            timestep_params["timeunit"] = timeunit
+        if n_timeunits is not None:
+            assert isinstance(n_timeunits, int)
+            timestep_params["numberOfTimeunits"] = n_timeunits
+        if end_of_week_day is not None:
+            assert end_of_week_day in {1, 2, 3, 4, 5, 6, 7}, "Invalid end_of_week_day, should be in [1, 2, 3, 4, 5, 6, 7]"
+            timestep_params["endOfWeekDay"] = end_of_week_day
+
+    def get_resampling_params(self):
+        """
+        Gets the time series resampling parameters for the time series forecasting task. This returns a reference to the
+        time series resampling parameters, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the resampling parameters
+        :rtype: dict
+        """
+        return self.mltask_settings["preprocessing"]["timeseriesSampling"]
+
+    def set_numerical_interpolation_params(self, method=None, constant=None):
+        """
+        Sets the time series resampling numerical interpolation parameters
+
+        :param method: Interpolation method. Valid values are: NEAREST, PREVIOUS, NEXT, LINEAR, QUADRATIC, CUBIC, CONSTANT
+        :type method: str
+        :param constant: number of time units within a time step
+        :type constant: object
```

Comment:
Not sure I get this param
- the name?
- looks like it should be a number but the type is `object`?

(multiple instances)